src/Promise.php

```diff
@@ -61,19 +60,15 @@ public function wait($unwrap = true)
     {
         $this->waitIfPending();
 
-        $inner = $this->result instanceof PromiseInterface
-            ? $this->result->wait($unwrap)
-            : $this->result;
-
+        if ($this->result instanceof PromiseInterface) {
+            return $this->result->wait($unwrap);
```

Comment:
when $unwrap is false, the old code did not return anything at all (= null). Now it returns what the inner promise returns. I don't think it matters though.